### ISSUE TITLE
Alt+{letter} menu shortcuts and arrow key navigation don't work on Windows and Linux

### ISSUE THREAD

**Issue by jmm** [atom/atom#4057]
The normal Windows `Alt+{letter}` menu shortcuts don't work in Atom. Holding `Alt` highlights the relevant letter in each menu name, but then pressing the letter doesn't open the menu. For example, `Alt+F` should open the File menu, but instead the Key Binding Resolver shows it invoking the following:

```
editor:move-to-end-of-word    atom-text-editor    resources\app\keymaps\emacs.json
```

This is a vanilla install of Atom 0.141.0 in Windows 7 Home Premium x64. I haven't opted in to enabling anything Emacs related.

Pressing Alt also doesn't focus the menu bar and allow it to be navigated by arrow keys like it normally does in Windows.

I'm reporting this per [this suggestion](https://discuss.atom.io/t/alt-key-menu-bar-interaction-overrides-shortuts/9192/5)

**Comment by Pajamaman** [comment]
+1 emacs behavior should not be default, especially for keys related to operating menus.

**Comment by kolAflash** [comment]
Workaround:

Edit -> Prerferences -> Keybindings -> "your keymap file"
keymap.cson
```
'atom-text-editor':
#  'alt-h': 'editor:delete-to-beginning-of-word'
  'alt-h': 'unset!'
#  'alt-f': 'editor:move-to-end-of-word'
  'alt-f': 'unset!'
```

**Comment by linquize** [comment]
Why Ubuntu Unity does not have such problem?

**Comment by kolAflash** [comment]
@linquize 
Because on Unity the menu is not part of Atom, but of the Unity menubar. So Unity is handling the shortcuts and not Atom itself. Principally the same thing which happends on MacOSX.

**Comment by poelzi** [comment]
This is a quite annoying bug. When you are new to a editor you often look into the wrong menu at first, having no left/right key there means mouse switching :(

**Comment by nycdotnet** [comment]
This is also an accessibility issue as well.

Even with the workaround, Atom still doesn't have the normal behavior on Windows which is to allow a single keypress of alt to be "sticky" and accept a second keypress after for the file menu.  People who can only physically press one key at a time will have difficulty with Atom's menus.

Thanks for posting your workaround, @kolAflash .

**Comment by pdl** [comment]
@kolAflash, thanks, that workaround works for me.

**Comment by joshas** [comment]
On Ubuntu 15.10 Atom 1.1.0 doesn't react to Alt+F if there is new untitled pane open and cursor is inside it. If I click on menu bar or any item in project sidebar - Alt+F works.
Same issue with Alt+H. Other menu shortcuts seems to work (Alt + E, V, S, I, P).

**Comment by garretwilson** [comment]
This is extremely annoying. Please remove these key bindings from the editor by default. On Windows it is very irritating for the menus not to come up. Sure, there are workarounds, but I think that accessing the menus is something that should work out of the box without a workaround.

**Comment by bekroogle** [comment]
I'm a little surprised that this issue is still unresolved. There are a lot of ways to navigate the cursor among words in a line using the keyboard (worst case scenario, just press the arrow keys a bunch), but with the default keybindings, I haven't found a single way of opening the file menu without using the mouse.

I quit using Atom for quite a few months in the past because this one behavior made me feel like the editor was still in some pre-alpha phase (it may have been at the time). I would imagine I'm not the only one who thought this was just a really glaring bug, instead of a counter-convention UX decision.

Anyway, thanks to @kolAflash for the workaround. I really want to like Atom, and that helped a lot!

**Comment by garretwilson** [comment]
Yes, @bekroogle , this is what happened to me as well. I tried Atom, and my initial thought was, "this must be early-stage software, because they haven't even figured out how to let Windows users access the menus correctly". The fact that I still can't use `Ctrl+F` to get the *File* menu is simply inexcusable. There is no other way about it.

If you want emacs key bindings, have the user install an emacs plugin!! Isn't that the philosophy of Atom in the first place? Why force Windows users to install key bindings most of them won't use, in the process taking away a fundamental UI usability convention? Absurd thinking that turns the Atom design philosophy on its head.

**Comment by MorrisJohns** [comment]
Another workaround that you can put into your own keymap.cson file (without needing to modify atom files):

```coffeescript
#Fix so that ALT-F and ALT-H activate menus properly on Windows
'atom-text-editor':
  'alt-f': 'native!'
  'alt-F': 'native!'
  'alt-h': 'native!'
  'alt-H': 'native!'
```

**Comment by garretwilson** [comment]
So I'm updating our Java development course for complete newbies. In the course we recommend an editor to our students. But we are throwing Atom out the window (no pun intended)---after all, it can't even work correctly on the Windows platform. The last thing new students to Java need is to muck around trying (and failing) to access the menus on their platform of choice. Even with the workarounds, there still remains #5998 for the arrow keys.

How embarrassing for Atom that it can't even get a simple menu system to work. I wish we could use it in our course, but we're tossing it and going with something else that has the basics down.

**Comment by neojp** [comment]
:+1:

**Comment by Neutrino-Sunset** [comment]
First of all I think the title of this issue needs to be updated to remove Windows from it, since the problem is glaring on Linux too, and as some people like to look down their noses at Windows any issues with Windows in the title may possibly not get the attention they deserve, (as seems to be happening here).

Secondly, while the workaround of unbinding the emacs keys works to make it possible to open the File menu there doesn't yet seem to be any workaround for getting the cursor keys to switch between menus.

In every other usage that comes to mind the cursor keys work as a group of four, so in this case for the up/down keys to move vertically through the menu but the left/right keys to do nothing is both highly frustrating and counter-intuitive. Is there really no fix for this?

P.S. Great editor.

**Comment by bilbonotry** [comment]
Honestly, it is really a pity that such a great editor is plagued by basic (in terms of functionality I mean, not code) bugs like this, corporate proxy handling or manual package install (shift-F10 for right-click also does not work).
Please consider priorizing such tasks.

**Comment by lemongrassnginger** [comment]
Until Shift-F10 is added to open the context menu, I'm trying to write a hack using atom commands and keybindings to trigger the browser's contextmenu event. Struggling though -  if anyone has any ideas about how to get something like this working, or could explain why this approach isn't worth pursuing, that'd be much appreciated! 

https://github.com/lemongrassnginger/context-menu-open

**Comment by damieng** [comment]
We have a few options here and I'll list them out separately so people can vote on them;

**Comment by damieng** [comment]
Disable emacs.cson by default on Windows and Linux to avoid the conflict. Users wanting them will have to bind them themselves.

**Comment by damieng** [comment]
Add a setting to determine if they are enabled or not with a suitable warning. Disable by default on Windows/Linux but no rebinding required.

**Comment by damieng** [comment]
Move emacs.cson to a package that is only installed by default on macOS. (I don't think we have this ability today however)

**Comment by damieng** [comment]
On further investigation it turns out the more common Emacs <kbd>ctrl</kbd> bindings were completely missing for Windows and Linux users and nobody reported it.

Given that I've made the <kbd>alt</kbd> and <kbd>alt</kbd>+<kbd>ctrl</kbd> emacs bindings Mac-only.

Anyone who actually does want a full set can simply enter them into their keybindings file via File > Keybindings.  Here's a sample list...

```cson

'atom-text-editor':
  'ctrl-a': 'editor:move-to-first-character-of-line'
  'ctrl-e': 'editor:move-to-end-of-line'
  'ctrl-k': 'editor:cut-to-end-of-line'
  'ctrl-p': 'core:move-up'
  'ctrl-n': 'core:move-down'
  'ctrl-b': 'core:move-left'
  'ctrl-f': 'core:move-right'
  'ctrl-P': 'core:select-up'
  'ctrl-N': 'core:select-down'
  'ctrl-F': 'core:select-right'
  'ctrl-B': 'core:select-left'
  'ctrl-h': 'core:backspace'
  'ctrl-d': 'core:delete'
  'alt-f': 'editor:move-to-end-of-word'
  'alt-ctrl-f': 'editor:move-to-next-subword-boundary'
  'alt-F': 'editor:select-to-end-of-word'
  'alt-ctrl-F': 'editor:select-to-next-subword-boundary'
  'alt-b': 'editor:move-to-beginning-of-word'
  'alt-ctrl-b': 'editor:move-to-previous-subword-boundary'
  'alt-B': 'editor:select-to-beginning-of-word'
  'alt-ctrl-B': 'editor:select-to-previous-subword-boundary'
  'alt-h': 'editor:delete-to-beginning-of-word'
  'alt-ctrl-h': 'editor:delete-to-beginning-of-subword'
  'alt-d': 'editor:delete-to-end-of-word'
  'alt-ctrl-d': 'editor:delete-to-end-of-subword'
```

**Comment by damieng** [comment]
Going to close this issue now as it's fixed in master and will go out in 1.12.  Emacs keys are not bound by default on Windows or Linux any more so ALT-F and ALT-H work fine for accessing the menu. (Many important emacs bindings were never present on these systems anyway).

The shift-F10, tapping ALT issues etc. are actually Electron issues being tracked at electron/electron#2504

**Comment by rbenson** [comment]
This issue has either not been fixed, or they been rolled back. `alt+F` and `alt+H` are still bound to editor:move-to-end-of-word and editor:delete-to-beginning-of-word by default

**Comment by damieng** [comment]
Working just fine here. What version of Atom are you on?

What are the contents of your File > Keymap

**Comment by rbenson** [comment]
Atom v1.10.2, and win 10

Keymap, added just before my last comment.

    'atom-text-editor':
       'alt-f': 'unset!'

**Comment by damieng** [comment]
Atom 1.10 is very old. The latest stable is 1.14.    I believe the fixes went into 1.12.

**Comment by rbenson** [comment]
Then the updater is broken because it says 1.10.2 is the latest stable version
